pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -3100,6 +3100,9 @@ type ClientSpec struct {
 	RemoveSecret bool `json:"removeSecret,omitempty"`
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Caps map[string]string `json:"caps"`
+	// CephX configures CephX key settings. More: https://docs.ceph.com/en/latest/dev/cephx/
+	// +optional
+	CephX ClusterCephxConfig `json:"cephx,omitempty"`
```

Comment:
The CephClient doesn't have a daemon associated, and the `ClusterCephxConfig` has a Daemon field. In the design doc, CephClient stands out in its simplicity, it should just have the raw cephx options.

```suggestion
	CephX CephxConfig `json:"cephx,omitempty"`
```